src/ClientGenerator/CSharpCodeGenerator.cs

```diff
@@ -455,12 +454,40 @@ protected override void AddGetGrainMethods(GrainInterfaceData iface, CodeTypeDec
                     add(@"
                         public static {0} GetGrain(System.String primaryKey)
                         {{
-                            return Cast(global::Orleans.CodeGeneration.GrainFactoryBase.MakeGrainReferenceInternal(typeof({0}), {1}, primaryKey));
+                            return global::Orleans.GrainFactory.GetGrain<{0}>(primaryKey);
                         }}");
 
                     add(@"
                         public static {0} GetGrain(System.String primaryKey, string grainClassNamePrefix)
                         {{
+                            return global::Orleans.GrainFactory.GetGrain<{0}>(primaryKey, grainClassNamePrefix);
+                        }}");
+                }
+
+                if (isDefaultKey)
+                {
+                    add(@"
+                        public static {0} GetGrain(long primaryKey)
+                        {{
+                            return Cast(global::Orleans.CodeGeneration.GrainFactoryBase.MakeGrainReferenceInternal(typeof({0}), {1}, primaryKey));
```

Comment:
This is still the old generated code. Had to keep it this way to deal with IGrain, since compiler cannot distinguish between GrainFactory.GetGrain<T> where T is IGrain  and  GrainFactory.GetGrain<T> where T is IGrainWithGuidKey.